ironfish/src/mining/stratum/stratumClient.ts

```diff
@@ -61,14 +56,14 @@ export class StratumClient {
     this.socket.on('data', (data) => void this.onData(data).catch((e) => this.onError(e)))
   }
 
-  start(): void {
+  async start(): Promise<void> {
     if (this.started) {
       return
     }
 
     this.started = true
     this.logger.info('Connecting to pool...')
-    void this.startConnecting()
+    await this.startConnecting()
```

Comment:
This should be voided, right? Why is this being awaited now?

Reply:
The change to await blocks the client from being started until it's finished connecting.

The StratumClient currently sends a 'subscribe' message automatically when it finishes connecting, but these changes have the miner making that call. It can only send that message once the client has connected, so it blocks until that's happened

Reply:
Right, but start() is not correlated to being connected. This just starts the connecting loop. What if it connects on the second time, instead of the first time? Start() will resolve but you won't be connected.

If things need to respond to becoming connected, then it should respond to an onConnected event that's fired rather than rely on start() which won't guarantee a connection.

Reply:
Good point, I didn't think through how this would resolve with connection retries